students/matt_casali/Lesson08/circle.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python3
+
+import math
+
+
+class Circle:
+    # Step 1
+    def __init__(self, radius):
+        self._radius = radius
+
+    @property
+    def radius(self):
+        return self._radius
+
+    @radius.setter
+    def radius(self, radius):
+        self._radius = radius
+
+    # Step 2 and 3
+    @property
+    def diameter(self):
+        return self._radius * 2
+
+    @diameter.setter
+    def diameter(self, diameter):
+        self._radius = diameter / 2
+
+    # Step 4
+    @property
+    def area(self):
+        area = (self._radius ** 2) * math.pi
+        return f"{area:.2f}"
+
+    # Step 5
+    @classmethod
+    def from_diameter(self, diameter):
+        return Circle(diameter / 2)
```

Comment:
with class methods you should use `cls` instead of `self`

then return `cls(diameter / 2)`